### PR TITLE
Creating standalone CMake build system for hafs-community/gfdl-tracker

### PR DESCRIPTION
**Pull request by BijuThomas-NOAA**
## Description
The changes in this merge will enable the vortextracker component to be built in a standalone mode with CMake and it can be easily transitioned vortextracker component to other systems outside HAFS. 

## Dependencies
- Dependent on: hafs-community/gfdl-tracker/pull/1
- Dependent on: hafs-community/gfdl-tracker/pull/2

### CODE REVIEW

File: .gitmodules
Comment (evankalina) [inline .gitmodules]: Is there a reason why the .gitmodules file needs to be changed here?

**Review by evankalina — COMMENTED** [review]
These changes look reasonable to me, and I think support for standalone builds is a nice feature. Did you have a chance to test these changes on any of the platforms? I assume if it works on one platform, it will work on others, since you basically just moved the existing machine-specific logic into the gfdl-tracker repository?

File: .gitmodules
Comment (BijuThomas-NOAA) [reply .gitmodules]: @evankalina, Thanks for bringing this to my attention. This is fixed.

**Comment by BijuThomas-NOAA** [comment]
> These changes look reasonable to me, and I think support for standalone builds is a nice feature. Did you have a chance to test these changes on any of the platforms? I assume if it works on one platform, it will work on others, since you basically just moved the existing machine-specific logic into the gfdl-tracker repository?

@evankalina, Tested it on Orion, WCOSS(Cray and Dell). The test is in progress on Jet.

**Review by evankalina — DISMISSED** [review]
Looks good to me!

**Comment by BinLiu-NOAA** [comment]
@BijuThomas-NOAA Please sync with the latest HAFS develop branch also point to the latest support/HAFS branch under hafs-community/gfdl-tracker.

File: .gitmodules
Comment (BinLiu-NOAA) [inline .gitmodules]: @BijuThomas-NOAA, please fix the indentation for this url line above. Thanks! Also make sure it points to the latest support/HAFS branch for sorc/hafs_vortextracker.fd.

File: sorc/build_vortextracker.sh
Comment (BinLiu-NOAA) [inline sorc/build_vortextracker.sh]: @BijuThomas-NOAA, I also made some minor changes in the gfdl-tracker through this PR: hafs-community/gfdl-tracker/pull/2. The build_all_cmake.sh has been moved from ush to src. So, once that PR is merged, please help to update it accordingly as well as point it to the latest commit of the support/HAFS branch of gfdl-tracker. Thanks!

File: .gitmodules
Comment (BinLiu-NOAA) [reply .gitmodules]: @BijuThomas-NOAA, please revert this empty space change, and also update the hafs_vortextracker.fd submodule by:
cd sorc/hafs_vortextracker.fd
git checkout support/HAFS
git pull
cd ../
git add  hafs_vortextracker.fd
git commit
then git push
After these, I think this PR is ready to be merged. Please feel free to let me know if you have any question. Thanks! Bin

Comment (BijuThomas-NOAA) [reply .gitmodules]: @BinLiu-NOAA,
I don't see the empty space locally when I open with vi
[submodule "gfdl-tracker"]
        path = sorc/hafs_vortextracker.fd
        url = https://github.com/hafs-community/gfdl-tracker.git
        branch = support/HAFS

File: sorc/build_vortextracker.sh
Comment (BijuThomas-NOAA) [reply sorc/build_vortextracker.sh]: @BinLiu-NOAA, Thanks for the comments. I think it is ready now for merging.

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me.

**Comment by BinLiu-NOAA** [comment]
@BijuThomas-NOAA, Thanks for addressing the review comments! Waiting for another review approval, then this PR is ready to merge. Thanks!

**Review by evankalina — APPROVED** [review]
Looks great!

**Pull request merged**
